src/test/java/edu/harvard/iq/dataverse/api/S3AccessIT.java

```diff
@@ -377,6 +382,236 @@ public void testDirectUpload() {
         String contentsOfDownloadedFile = downloadFile.getBody().asString();
         assertEquals(contentsOfFile, contentsOfDownloadedFile);
 
+        Response getFileData = UtilIT.getFileData(fileId, apiToken);
+        getFileData.prettyPrint();
+        getFileData.then().assertThat()
+                .statusCode(OK.getStatusCode())
+                .body("data.label", equalTo("file1.txt"))
+                .body("data.dataFile.filename", equalTo("file1.txt"))
+                .body("data.dataFile.contentType", equalTo("text/plain"))
+                .body("data.dataFile.filesize", equalTo(6));
+
+        Response deleteFile = UtilIT.deleteFileApi(Integer.parseInt(fileId), apiToken);
+        deleteFile.prettyPrint();
+        deleteFile.then().assertThat().statusCode(200);
+
+        AmazonS3Exception expectedException = null;
+        try {
+            s3localstack.getObjectAsString(BUCKET_NAME, keyInS3);
+        } catch (AmazonS3Exception ex) {
+            expectedException = ex;
+        }
+        assertNotNull(expectedException);
+        // 404 because the file has been sucessfully deleted
+        assertEquals(404, expectedException.getStatusCode());
+
+    }
+
+    @Test
+    public void testDirectUploadDetectStataFile() {
```

Comment:
@qqmyers please take a look at this test I added (mostly copied from the one above).

Reply:
This test is OK, and it passes, so I'm fine with including it. A better and shorter one would be to either submit the mimetype as application/octet-stream or even application/x-stata so that the new detection works during upload rather than requiring it to be triggered in a redetect call. I'm not sure that testing download helps here either - that's just part of the cut/paste I assume.

Reply:
In 614dc7e44c I'm passing application/octet-stream but file detection on initial upload doesn't seem to be working.

In that commit you'll see the comment "Should the contentType be application/x-stata-14 already?"

Later, after the redetect call, the content type is application/x-stata-14.

Am I testing it wrong?

Reply:
FileUtil line 433 is checking to make sure the base type is application/x-stata to start, which would happen if you send the .dta extension on the file (or if you sent application/x-stata to start). FWIW: This is mainly for efficiency/ to avoid checking all files for stata14 and to just test those that were getting missed/left as x-stata before.

Reply:
Thanks for talking this out today. I updated the tests to have the file end in .dta and file detection is working, like you said.